api/pkg/transformer/feast/feature_retriever.go

```diff
@@ -215,10 +215,11 @@ func (fr *FeastRetriever) newCall(featureTableSpec *spec.FeatureTable, columns [
 	}
 
 	return &call{
```

Comment:
Rather than creating `call` struct directly, I think it's better to have a constructor `NewCall` and abstract away `getFeatureTypeMapping` behind it. Additionally, `defaultValues`, `logger`, `statusMonitoringEnabled`, `valueMonitoringEnabled` can all be derived from `fr`

Reply:
updated